modules/optflow/samples/dis_opticalflow.cpp

```diff
@@ -0,0 +1,75 @@
+
+#include "opencv2/core/utility.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgproc/types_c.h"
+#include "opencv2/optflow.hpp"
+
+using namespace std;
+using namespace cv;
+using namespace optflow;
+
+static void help()
+{
+    printf("Usage: dis_optflow <video_file>\n");
+}
+
+int main(int argc, char **argv)
+{
+    VideoCapture cap;
+
+    if (argc < 2)
+    {
+        help();
+        exit(1);
+    }
+
+    cap.open(argv[1]);
+    if(!cap.isOpened())
+    {
+        printf("ERROR: Cannot open file %s\n", argv[1]);
+        return -1;
+    }
+
+    Mat prevgray, gray, rgb, frame;
+    Mat flow, flow_uv[2];
+    Mat mag, ang;
+    Mat hsv_split[3], hsv;
+    char ret;
+
+    namedWindow("flow", 1);
+    namedWindow("orig", 1);
+
+    Ptr<DenseOpticalFlow> algorithm = createOptFlow_DIS(DISOpticalFlow::PRESET_MEDIUM);
+
+    while(true)
+    {
+        cap >> frame;
+        if (frame.empty())
+            break;
+
+        cvtColor(frame, gray, CV_BGR2GRAY);
```

Comment:
You can use `COLOR_BGR2GRAY` (like line 64 below) and remove "opencv2/imgproc/types_c.h" dependency